src/ibek/support_cmds/commands.py

```diff
@@ -56,51 +56,80 @@ class AptWhen(str, Enum):
-        AptWhen.both, help="which container build stage to install in"
-    ),
-    runtime: bool = typer.Option(False, help="install list of runtime packages"),
-):
+def _install_debs(debs: List[str], args) -> None:
     """
-    Install debian packages into the container. If they have an http:// or https://
+    Install a list of debian packages.
+
+    If they have an http:// or https://
     prefix then they will be downloaded and installed from file.
     """
-    if runtime and not GLOBALS.NATIVE:
-        print("skipping runtime install in cross-compile environment")
-        return
-
     temp = Path("/tmp")
-
-    if (only is AptWhen.run) or (only is AptWhen.both):
-        add_list_to_file(RUNTIME_DEBS, debs)
-    if only is AptWhen.run:
-        return
-
-    if runtime and RUNTIME_DEBS.exists():
-        debs += RUNTIME_DEBS.read_text().split()
-
     for i, pkg in enumerate(debs):
         if pkg.startswith("http://") or pkg.startswith("https://"):
             pkg_file = temp / pkg.split("/")[-1]
             subprocess.call(["wget", pkg, "-O", str(pkg_file)])
             debs[i] = str(pkg_file)
 
+    if len(debs) == 0:
+        print("no packages to install")
+        return
+
+    print("installing packages: ", debs)
+
     sudo = "sudo" if os.geteuid() != 0 else ""
     command = (
         f"{sudo} apt-get update && {sudo} apt-get upgrade -y && "
         f"{sudo} apt-get install -y --no-install-recommends "
         + " ".join(debs)
-        + " ".join(ctx.args)
+        + " ".join(args)
     )
     exit(subprocess.call(["bash", "-c", command]))
 
 
+@support_cli.command(
+    context_settings={"allow_extra_args": True, "ignore_unknown_options": True}
+)
+def runtime_apt_install(
+    ctx: typer.Context,
+    debs: List[str] = typer.Argument(None, help="list of debian packages to install"),
+):
+    """
+    Install debian packages into the runtime container. Use the list of
+    packages in the RUNTIME_DEBS_debs file and also accept additional packages
+    as arguments.
+    """
+    if not GLOBALS.NATIVE:
+        print("skipping runtime install in cross-compile environment")
+        return
+
+    if RUNTIME_DEBS.exists():
+        debs += RUNTIME_DEBS.read_text().split()
+
+    _install_debs(debs, ctx.args)
+
+
+@support_cli.command(
+    context_settings={"allow_extra_args": True, "ignore_unknown_options": True}
+)
+def apt_install(
+    ctx: typer.Context,
+    debs: List[str] = typer.Argument(None, help="list of debian packages to install"),
+    only: AptWhen = typer.Option(
+        AptWhen.both, help="which container build stage to install in"
+    ),
+):
+    """
+    Install debian packages into the developer container.
+    Also add to the list of packages to install in the runtime container.
+    """
+    if (only is AptWhen.run) or (only is AptWhen.both):
+        add_list_to_file(RUNTIME_DEBS, debs)
+    if only is AptWhen.run:
+        return
+
+    _install_debs(debs, ctx.args)
+
+
```

Comment:
I think this is fine, but just to question everything, what about:

```python
def apt_install(
    ctx: typer.Context,
    debs: List[str] = typer.Argument(None, help="list of debian packages to install"),
):
    # Install packages "now", i.e. whatever stage of the Dockerfile this is called in

def add_runtime_packages(
    ctx: typer.Context,
    debs: List[str] = typer.Argument(None, help="list of debian packages to install"),
):
    # Add packages to RUNTIME_DEBS

def apt_install_runtime_packages(
    ctx: typer.Context,
):
    # Install packages from RUNTIME_DEBS "now", i.e. whatever stage of the Dockerfile this is called in

```

This might mean some duplication in lists of packages, but it would be more explicit and less code. Is there an example where we are doing `apt-install --when=both`, I can't find any.

Reply:
I'm making sweeping breaking changes right now so happy to do this if it gets more votes. @coretl ?

Reply:
I don't think we have had a need for =both yet. It could easily come up in later support modules but I think two lines to describe this is clearer anyway.

So I have talked myself into adopting Gary's suggetion.

Reply:
So I guess we would now do things like this in `install.sh`:
```bash
ibek apt-install pkg1-dev pkg2-dev
ibek add-runtime-packages pkg1 pkg2
```
That looks clean enough to me. I will vote for this too.

Reply:
:+1: 
going ahead with this idea